[Lin Leyi] iP

lleyi0606 opened the pull request:
## Dukie

> “It takes as much energy to wish as it does to plan.” – Eleanor Roosevelt ([Source](https://www.brainyquote.com/quotes/eleanor_roosevelt_379411))

Your one-stop solution to record and manage your daily tasks. It's

- text-based
- efficient 
- straightforward to use! 

You just need to, 

1. download it from [here](https://github.com/lleyi0606/ip)
2. open it with your favourite code editor
3. input your tasks
4. sit back and let Dukie ~~do~~ remember them 🐥
 
### Features
- [ ] managing different types of tasks (todo, event, deadline) 
- [ ] sorting by date
- [ ] reminders _(coming soon)_ 

If you are a Java Programmer too, you can use it to practice programming! 
```ruby
public static void main(String[] args) {
        new Duke("data/duke.txt").run();
}
```

Commit:
Add Gradle support

Commit:
Level-1 Commit

Commit:
Level-2 Commit

Commit:
Commit Level-3

Commit:
commit level-4

Commit:
semi-automate testing

Commit:
added exceptions

Commit:
implemented delete

Commit:
add write to duke.txt function

Commit:
create branch and re-commit

Commit:
add store date as LocalDate object and "schedule" function

Commit:
fix merge conflicts

Commit:
add Ui, Storage, Parser, TaskList classes

Commit:
organise classes into package duke

Commit:
edit file path of duke

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'

Commit:
Use gradle to add JUnit Tests

Commit:
Add class header and some method documentations

Commit:
fixed code to comply with coding standards

Commit:
add find

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
merge branch 'branch-Level-9'

feliciagan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                        ui.showMessage(e.getMessage());
+                    }
+                    break;
+                default:
```
Case clauses should not be indented

feliciagan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,53 @@
+            return userInput.substring(userInput.indexOf("/") + 4);
+        }
+        return "";
+    }
```
If possible, could include javadoc comment for this method (and other methods in the other files)

feliciagan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,142 @@
+Task object that gets created by the user
+*/
+public class Task {
+    private boolean done;
```
Perhaps a more boolean-sounding name like isDone?

feliciagan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,142 @@
+        return this.taskName;
+    }
+
+    public String isDoneString() {
```
Maybe a method name like getIsDoneString()? (since names representing methods must be verbs)

feliciagan reviewed (commented):
looks good!

phualiting commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,142 @@
+                this.taskType = TaskType.EVENT;
+            } else {
+                this.taskType = TaskType.DEADLINE;
+            }
```
Would it be better to remove the inner if else block, so that it is not checking for deadline and event twice, and will be clearer?

phualiting commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,142 @@
+/*
+Task object that gets created by the user
+*/
+public class Task {
```
I think it would be clearer if you add additional Todo, Deadline and Event classes that extend Task, as I feel the checking of tasktype can be quite confusing?

phualiting commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+                    //ui.showList(foundTasks);
+                    ui.showFoundTasks(foundTasks, keyword);
+                case "todo":
+                case "deadline":
```
Are these missing the implementation?

phualiting commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+            userInput = myObj.nextLine();  // Read user input
+            String command = Parser.parseUserInput(userInput);
+
+            switch (command) {
```
I think this would be a good place to use enums?

phualiting reviewed (commented):
Overall, I think the code mostly follows the coding standard and method names are easy to understand. It would be clearer if todo, deadline and event were separated into different classes.

Commit:
implement basic gui with javafx

Commit:
create GUI to Duke

Commit:
cleaned up

Commit:
Improve code quality

All main logic are in Duke, resulting in poor code quality.

Lack of OOP and violation of information encapsulation. Also might be more difficult to catch errors.

Let's create separate Command classes to separate the code that is used for each one.

This makes the Duke class cleaner and demonstrates SLAP.

lleyi0606 committed:
Merge pull request #2 from lleyi0606/branch-A-CodeQuality

Improve code quality

Commit:
Add assertions

Lack of assertions.

Code makes assumptions about certain properties but this assumption is not checked. Hence could lead to errors that are undetected.

Let's add assertions to verify if crucial assumptions are indeed true.

This is such that if these assumptions are not met, we can readily spot the bug in the code and fix it before user uses it.

Commit:
Add snooze command

lleyi0606 committed:
Merge pull request #3 from lleyi0606/branch-A-Assertions

Add assertions

lleyi0606 committed:
Merge pull request #4 from lleyi0606/branch-B-Snooze

Add snooze command

Commit:
Update README and javadocs

Commit:
update README

Commit:
Fix Storage file creation
